noggin/security/ipa.py

```diff
@@ -35,6 +37,15 @@ class Client(IPAClient):
     TODO: send this upstream.
     """
 
+    def group_find(self, *args, **kwargs):
+        """
+        Find a group. A wrapped method to allow filtering of groups to be hidden from the user.
+
+        TODO: this needs to be retained when we migrate to the newer Client class.
+        """
+        kwargs.setdefault("not_in_group", current_app.config.get('HIDDEN_GROUPS'))
```

Comment:
If you do it that way, a call to `group_find(not_in_group="blah")` will only be excluding groups that are in the `blah` group, not the one in the config. That's why I suggested something different in my last review. I know the `setdefault()` method is not very obvious to use, feel free to do it differenly if you prefer.

Reply:
Yeah, it wasn't so much that. It was that `not_in_group` can only take one group. If we want to still allow a different exclusion somewhere in our app, that means I'll need to make the call twice and add those groups together. that seems a bit weird to me becuase they may not be mutually exclusive.

Reply:
Do you see us using `not_in_group` anywhere else?

Reply:
Actually, `not_in_group` can take a list of groups, I just tested it. Did it not work for you? I don't however think we're ever going to use a list of groups here, so I guess a single value is OK.

However I think it would be nice to remove the "hidden-groups" group from the output, and for that I think we can set the `posix` keyword argument. I'll only output posix groups, and I think that's the only ones we need.